src/main/java/com/commercetools/sync/sdk2/states/utils/StateReferenceResolutionUtils.java

```diff
@@ -0,0 +1,107 @@
+package com.commercetools.sync.sdk2.states.utils;
+
+import static com.commercetools.sync.sdk2.commons.utils.SyncUtils.getResourceIdentifierWithKey;
+
+import com.commercetools.api.models.state.State;
+import com.commercetools.api.models.state.StateDraft;
+import com.commercetools.api.models.state.StateDraftBuilder;
+import com.commercetools.api.models.state.StateReference;
+import com.commercetools.api.models.state.StateResourceIdentifier;
+import com.commercetools.api.models.state.StateResourceIdentifierBuilder;
+import com.commercetools.sync.sdk2.commons.utils.ReferenceIdToKeyCache;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+
+/**
+ * Util class which provides utilities that can be used when syncing resources from a source
+ * commercetools project to a target one.
+ */
+public final class StateReferenceResolutionUtils {
+
+  /**
+   * Returns an {@link java.util.List}&lt;{@link StateDraft}&gt; consisting of the results of
+   * applying the mapping from {@link State} to {@link StateDraft} with considering reference
+   * resolution.
+   *
+   * <table>
+   *   <caption>Mapping of Reference fields for the reference resolution</caption>
+   *   <thead>
+   *     <tr>
+   *       <th>Reference field</th>
+   *       <th>from</th>
+   *       <th>to</th>
+   *     </tr>
+   *   </thead>
+   *   <tbody>
+   *     <tr>
+   *        <td>transitions</td>
+   *        <td>{@link java.util.List}&lt;{@link StateReference}&gt;</td>
+   *        <td>{@link java.util.List}&lt;{@link StateReference}&gt; (with key replaced with id field)</td>
```

Comment:
StateReference is mapped to StateResourceIdentifier

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/1035/commits/e8866213c682dd55c8ae995ae6dbfac89c4ba3c0